### PR TITLE
relay splitter daemon + pebble persistence

### PR DESCRIPTION
**Pull request by brianolson**
stack PR on top of #807 
Add Pebble db based persister
clean up tests in events/

### CODE REVIEW

File: events/pebblepersist.go [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+
+	if seq < 0 {
+		// persist with longer key {prev 8 byte key}{time}{int32 extra counter}
+		pp.prevSeqExtra++
+		var key [20]byte
+		binary.BigEndian.PutUint64(key[:8], uint64(pp.prevSeq))
+		binary.BigEndian.PutUint64(key[8:16], nowMillis)
+		binary.BigEndian.PutUint32(key[16:], pp.prevSeqExtra)
+
+		err = pp.db.Set(key[:], blob, pebble.Sync)
```
Comment (whyrusleeping) [inline events/pebblepersist.go:51]: error ignored

Comment (whyrusleeping) [reply events/pebblepersist.go:51]: oh, we should just ignore -1 sequence number events (info, error, etc)

File: events/pebblepersist.go [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+func (pp *PebblePersist) SetEventBroadcaster(broadcast func(*XRPCStreamEvent)) {
+	pp.broadcast = broadcast
+}
+
+func (pp *PebblePersist) GetLast(ctx context.Context) (*XRPCStreamEvent, error) {
+	iter, err := pp.db.NewIterWithContext(ctx, &pebble.IterOptions{})
+	if err != nil {
+		return nil, err
+	}
+	ok := iter.Last()
```
Comment (whyrusleeping) [inline events/pebblepersist.go:130]: is the iter ordered by default?

File: events/pebblepersist.go [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+	expired := uint64(nowMillis - retention.Milliseconds())
+	iter, err := pp.db.NewIterWithContext(ctx, &pebble.IterOptions{})
+	if err != nil {
+		return err
+	}
+	defer iter.Close()
+	todel := make(chan []byte, 100)
+	go func() {
+		for xkey := range todel {
+			pp.db.Delete(xkey, nil)
```
Comment (whyrusleeping) [inline events/pebblepersist.go:168]: is there a batch delete option? or a range delete?

File: pds/server.go [hunk 4]
```diff
@@ -456,18 +457,7 @@ func (s *Server) HandleResolveDid(c echo.Context) error {
...
-	UpdatedAt   time.Time
-	DeletedAt   gorm.DeletedAt `gorm:"index"`
-	Handle      string         `gorm:"uniqueIndex"`
-	Password    string
-	RecoveryKey string
-	Email       string
-	Did         string `gorm:"uniqueIndex"`
-	PDS         uint
-}
+type User = pdsdata.User
```
Comment (whyrusleeping) [inline pds/server.go:460]: feels like a superfluous change to include in this PR

File: events/pebblepersist.go (hunk 2 above)
Comment (brianolson) [reply events/pebblepersist.go:130]: Yup

File: pds/server.go (hunk 4 above)
Comment (brianolson) [reply pds/server.go:460]: Yeah, but I was working on the unit tests, and the tests were in a different package which was weird, so I fixed that and then had to move a data definition to fix a circular dependency.

File: events/pebblepersist.go (hunk 3 above)
Comment (brianolson) [reply events/pebblepersist.go]: Yes! rushed and didn't see it the first time. done.

**Comment by brianolson** [comment]
TODO: I think we still have a TODO of using the pebble store's "get last entry" to set the sequence we fetch `?since={}` from the upstream firehose

**Comment by brianolson** [comment]
merged as #818

**Pull request closed without merging**